driver/vhci_ude/vhci_dev.h

```diff
@@ -13,7 +13,7 @@ typedef struct
 	ULONG		n_max_ports;
 	WDFQUEUE	queue;
 	struct _ctx_vusb	**vusbs;
-	WDFWAITLOCK	lock;
+	WDFSPINLOCK	spin_lock;
```

Comment:
Impressive try. Currently performance does not matter, so I didn't really care if spinlock is possible. I'll check if a spinlock is Ok. By the way, I'm curious about what's your reason for using spinlock.

Reply:
Well, it all boils down to resolving error states, when i could not replace driver because of blocking uninstall. Then i enabled the 'verifier' and 'debug' mode. Debugging 'minidumps' pointed me towards current proposal.

Reply:
And regarding splinlocks, if i remember correctly, there were BSOD-s where 'waitlocks' shoud bot be used in certain 'irq-levels'.